### ISSUE TITLE
confused with consistency loss

### ISSUE THREAD

**Issue by zyf12389** [PeikeLi/Self-Correction-Human-Parsing#23]
Thanks for your excellent work.
My question is as follows:
parsing_pre and edge_pre are all obtained through argmax in the code of consistency loss, are they backpropagable? If not, what positive effects have they brought?

Thanks.

**Comment by PeikeLi** [comment]
We have noticed the effect you mentioned. This loss term aims to ensure that the predicted parsing result
matches the predicted edge. The edge_pre should not be passed through argmax op. A modified pytorch implementation will be updated toward this issue.
